src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/DungeonFeatures.kt

```diff
@@ -653,19 +681,46 @@ object DungeonFeatures {
         }
         if (event.packet is S29PacketSoundEffect) {
             val packet = event.packet
+            val sound = packet.soundName
+            val pitch = packet.pitch
+            val volume = packet.volume
             if (Skytils.config.disableTerracottaSounds && isInTerracottaPhase) {
-                val sound = packet.soundName
-                val pitch = packet.pitch
-                val volume = packet.volume
                 if (sound == "game.player.hurt" && pitch == 0f && volume == 0f) event.isCanceled = true
                 if (sound == "random.eat" && pitch == 0.6984127f && volume == 1f) event.isCanceled = true
             }
+            if (sound == "random.bow" && volume == 1f && arrowCount != -1) {
+                val extraAttr = ItemUtil.getExtraAttributes(mc.thePlayer.heldItem)
+                if (extraAttr != null) {
+                    val level = when {
+                        mc.thePlayer.isSneaking -> 0
+                        else -> extraAttr.getCompoundTag("enchantments")?.getInteger("infinite_quiver")?.times(3) ?: 0
+                    }
+                    val randomNum = Random.nextInt(0, 100)
+                    if (level <= randomNum + 1 && arrowCount > 0) {
+                        arrowCount -= 1
+                    }
+                }
+            }
         }
     }
 
     @SubscribeEvent
     fun onGuiOpen(event: GuiOpenEvent) {
         rerollClicks = 0
+        val old = mc.currentScreen
+
+        if (old is GuiChest) {
+            val chest = old.inventorySlots as? ContainerChest ?: return
```

Comment:
Just for myself already, shouldn't return if code gets added after it

Reply:
Also another note for myself, could make it arrowCount = -2 if Slimeballs or similar are detected and just make it not work until they get removed

Reply:
hypixel loves consistent behavior! §r§aJax forged §r§fFlint Arrow§r§8 x2816 §r§afor §r§613,657.6 Coins§r§a!§r

Reply:
people don't have max quiver

Reply:
make it show mid-run